src/BSON/Binary.c

```diff
@@ -142,6 +152,32 @@ static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
 	efree(intern);
 } /* }}} */
 
+zend_object* php_phongo_binary_create_object(zend_class_entry *class_type TSRMLS_DC) /* {{{ */
+{
+        php_phongo_binary_t *intern;
+
+        intern = (php_phongo_binary_t *)ecalloc(1, sizeof(php_phongo_binary_t)+zend_object_properties_size(class_type));
+
+        zend_object_std_init(&intern->std, class_type TSRMLS_CC);
+        object_properties_init(&intern->std, class_type);
+
+        intern->std.handlers = &php_phongo_handler_binary;
+
+        return &intern->std;
+} /* }}} */
+#else
+static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
+{
+        php_phongo_binary_t *intern = (php_phongo_binary_t*)object;
+
+        zend_object_std_dtor(&intern->std TSRMLS_CC);
+
+        if (intern->data) {
+                efree(intern->data);
+        }
+        efree(intern);
```

Comment:
According to https://wiki.php.net/phpng-upgrading#zend_object_handlers this efree() should be removed